pkg/operator/controllers/monitoring/monitoring_controller_test.go

```diff
@@ -156,6 +200,45 @@ prometheusK8s:
       resources:
         requests:
           storage: 100Gi
+`,
+		},
+		{
+			name: "enabled and we want to disable",
```

Comment:
IIUC, this test case covers the scenario where Prometheus was initially deployed with a `volumeClaimTemplate` spec, and then the feature gate is disabled, the entire `volumeClaimTemplate` block will be removed from the statefulset template. 

I am not sure that will work since `volumeClaimTemplate` is immutable, even though `volumeClaimTemplate.spec.resources` is mutable. I tested by deploying this sts to the shared cluster, and when attempting to remove the `volumeClaimTemplate`, Kubernetes rejected the update.

```yaml


apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: nginx
  labels:
    kubernetes.io/name: nginx-sts
spec:
  serviceName: nginx-sts
  selector:
    matchLabels:
      kubernetes.io/name: nginx-sts
  template:
    metadata:
      labels:
        kubernetes.io/name: nginx-sts
    spec:
      containers:
      - name: nginx
        image: nginx
        ports:
        - containerPort: 80
          name: http
        volumeMounts:
        - name: www
          mountPath: /usr/share/nginx/html
  volumeClaimTemplates:
  - metadata:
      name: www
    spec:
      accessModes: ["ReadWriteOnce"]
      resources:
        requests:
          storage: 1Gi
```

Error message returned by `kubectl edit`:
```
 # statefulsets.apps "nginx" was not valid:
 # * spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

Reply:
@ihcsim this code doesn't remove `volumeClaimTemplate` from the statefullSet, it removes it from the configmap that Cluster Monitoring Operator watches. I think in this case the CMO will re-create the prom statefulset

Reply:
Cool - thanks for the explanation.